spec/infinispan_cluster_spec.js

```diff
@@ -30,11 +33,467 @@ describe('Infinispan cluster client', function() {
       .catch(t.failed(done)).finally(done);
   });
 
+  it('can put a key/value pair using different clients', function(done) {
```

Comment:
OK, understood :) Thanks a lot for detailed explanation. So then, I have removed the multi-client tests as there is not purpose in them. I have removed the tests for iterators as well because I have seen that you will add them in your pull request for fix of iterators. 
I have kept the tests for the listeners but only with one client and actually at the moment it would be a duplicate code as we have the same for the local tests. The issues [JDG-186](https://issues.jboss.org/browse/JDG-186) and [JDG-187](https://issues.jboss.org/browse/JDG-187) still appear with one client in clustered mode. I think the JDG-186 appears again because of round robin - the listener is tried to be removed from the other node rather than one where the listener was added. 

As soon as your PRs will be integrated (the ones for numbers returning as well as the iterators fix), I'll rebase this PR, as well as will get rid of duplicate listener tests, and then it can be ready for integration.